### PR TITLE
Use spring data criteria

### PR DESCRIPTION
**Pull request by calvinlu3**
This fixes https://github.com/oncokb/oncokb/issues/3466

- Regenerated the codebase with ElasticSearch removed and replaced with [Spring Data Criteria Queries](https://www.jhipster.tech/entities-filtering/)
- Updated `SearchService.java` to use spring criteries
- For each `EntityQueryService` class the main changes were:
  - Adding a `findBySearchQuery` method
  - Jhipster only generated code for searching by DrugBrand id field in the `createSpecification` method. I added the ability to search by DrugBrand name by following the generated code example.

### CODE REVIEW

File: src/main/java/org/mskcc/oncokb/curation/service/AlterationQueryService.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+/**
+ * Service for executing complex queries for {@link Alteration} entities in the database.
+ * The main input is a {@link AlterationCriteria} which gets converted to {@link Specification},
+ * in a way that all the filters must apply.
+ * It returns a {@link List} of {@link Alteration} or a {@link Page} of {@link Alteration} which fulfills the criteria.
+ */
+@Service
+@Transactional(readOnly = true)
+public class AlterationQueryService extends QueryService<Alteration> {
```
Comment (zhx828) [inline src/main/java/org/mskcc/oncokb/curation/service/AlterationQueryService.java:27]: Are these QueryService all auto generated?

File: src/main/java/org/mskcc/oncokb/curation/service/FdaSubmissionQueryService.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+     */
+    protected Specification<FdaSubmission> createSpecification(FdaSubmissionCriteria criteria) {
+        Specification<FdaSubmission> specification = Specification.where(null);
+        if (criteria != null) {
+            // This has to be called first, because the distinct method returns null
+            if (criteria.getDistinct() != null) {
+                specification = specification.or(distinct(criteria.getDistinct()));
+            }
+            if (criteria.getId() != null) {
+                specification = specification.or(buildRangeSpecification(criteria.getId(), FdaSubmission_.id));
```
Comment (zhx828) [inline src/main/java/org/mskcc/oncokb/curation/service/FdaSubmissionQueryService.java:98]: What is `FdaSubmission_`?

**Review by zhx828 — COMMENTED** [review]
I don't think there is much to review, mostly auto generated?

File: src/main/java/org/mskcc/oncokb/curation/service/AlterationQueryService.java (hunk 1 above)
Comment (calvinlu3) [reply src/main/java/org/mskcc/oncokb/curation/service/AlterationQueryService.java:27]: Yeah, they were generated.

File: src/main/java/org/mskcc/oncokb/curation/service/FdaSubmissionQueryService.java (hunk 2 above)
Comment (zhx828) [reply src/main/java/org/mskcc/oncokb/curation/service/FdaSubmissionQueryService.java:98]: Reading through it, is metamodal generated in the runtime?

Comment (calvinlu3) [reply src/main/java/org/mskcc/oncokb/curation/service/FdaSubmissionQueryService.java:98]: Its during build time and is located under the `target` folder

**Review by zhx828 — APPROVED** [review]

**Pull request merged**